tests/shell/src/main.c

```diff
@@ -316,6 +317,32 @@ static void test_set_root_cmd(void)
 	test_shell_execute_cmd("shell colors on", 0);
 }
 
+static void test_shell_fprintf(void)
+{
+	const char expect[] = "testing 1 2 3";
+	const struct shell *shell;
+	const char *buf, *start;
+	size_t size;
+
+	shell = shell_backend_record_get_ptr();
+	zassert_not_null(shell, "Failed to get shell");
+
+	buf = shell_backend_record_get_output(shell, &size);
+	shell_fprintf(shell, SHELL_VT100_COLOR_DEFAULT, "testing %d %s %c",
+		      1, "2", '3');
+	buf = shell_backend_record_get_output(shell, &size);
+	zassert_true(size >= sizeof(expect), "Expected size > %u, got %d",
+		     sizeof(expect), size);
+
+	/*
+	 * There are prompts and various ANSI characters in the output, so just
+	 * check that the string is in there somewhere.
+	 */
+	start = buf + size - sizeof(expect);
+	zassert_true(strstr(buf, expect),
```

Comment:
shouldn't it be `strstr(start, expect)`?

Reply:
Actually I should just drop that var